From c930a9c8565a2aba7d803d991a62747ce4ebf7b6 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Tue, 15 Jul 2014 16:57:59 +1000
Subject: [PATCH] metadata: properly handle media search type if we already
 have an extended inetref.

(cherry picked from commit 30b03cc2b86e949b434b7afc207f12d35e6c6d87)
---
 mythtv/libs/libmythmetadata/metadatacommon.cpp |   31 ++++++++++++++++++++++++
 mythtv/libs/libmythmetadata/metadatacommon.h   |    2 +-
 2 files changed, 32 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythmetadata/metadatacommon.cpp b/mythtv/libs/libmythmetadata/metadatacommon.cpp
index 950c373..02e889b 100644
--- a/mythtv/libs/libmythmetadata/metadatacommon.cpp
+++ b/mythtv/libs/libmythmetadata/metadatacommon.cpp
@@ -423,6 +423,37 @@ ArtworkList MetadataLookup::GetArtwork(VideoArtworkType type) const
     return ret;
 }
 
+LookupType MetadataLookup::GetSubtype() const
+{
+    if (m_inetref.isEmpty() ||
+        m_inetref == MetaGrabberScript::CleanedInetref(m_inetref) ||
+        (m_subtype != kProbableMovie && m_subtype != kProbableTelevision &&
+         m_subtype != kProbableGenericTelevision))
+    {
+        // can't determine subtype from inetref
+        return m_subtype;
+    }
+
+    MetaGrabberScript grabber =
+        MetaGrabberScript::GetGrabber(m_subtype == kProbableMovie ?
+                                        kGrabberMovie : kGrabberTelevision,
+                                      this);
+
+    if (!grabber.IsValid())
+    {
+        return m_subtype;
+    }
+    switch (grabber.GetType())
+    {
+        case kGrabberMovie:
+            return kProbableMovie;
+        case kGrabberTelevision:
+            return kProbableTelevision;
+        default:
+            return m_subtype;
+    }
+}
+
 void MetadataLookup::toMap(InfoMap &metadataMap)
 {
     metadataMap["filename"] = m_filename;
diff --git a/mythtv/libs/libmythmetadata/metadatacommon.h b/mythtv/libs/libmythmetadata/metadatacommon.h
index c2666a6..604ba08 100644
--- a/mythtv/libs/libmythmetadata/metadatacommon.h
+++ b/mythtv/libs/libmythmetadata/metadatacommon.h
@@ -273,7 +273,7 @@ class META_PUBLIC MetadataLookup : public QObject, public ReferenceCounter
     // GETS
 
     MetadataType GetType() const { return m_type; };
-    LookupType GetSubtype() const { return m_subtype; };
+    LookupType GetSubtype() const;
     QVariant GetData() const { return m_data; };
     LookupStep GetStep() const { return m_step; };
     bool GetAutomatic() const { return m_automatic; };
-- 
1.7.10.2

